modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/APITagVisibilityByRoleTestCase.java

```diff
@@ -20,218 +20,137 @@
-    private static final long WAIT_TIME = 45 * 1000;
-
-    @Factory(dataProvider = "userModeDataProvider")
-    public APITagVisibilityByRoleTestCase(TestUserMode userMode) {
-        this.userMode = userMode;
-    }
+    private String providerName;
+    private RestAPIPublisherImpl apiPublisherClientCarbonSuperUser1;
+    private String publicApiId;
+    private String restrictedApiId;
+    private RestAPIStoreImpl apiStoreClientAllowedUser;
+    private RestAPIStoreImpl anonymousRestAPIImpl;
 
     @BeforeClass(alwaysRun = true)
-    public void setEnvironment() throws Exception {
-        super.init(userMode);
-
-        publisherURLHttp = getPublisherURLHttp();
+    public void initialize() throws Exception {
+        super.init();
+        endpointUrl = backEndServerUrl.getWebAppURLHttp() + API_END_POINT_POSTFIX_URL;
         storeURLHttp = getStoreURLHttp();
-        endpointUrl = backEndServerUrl.getWebAppURLHttp() + "am/sample/calculator/v1/api/add";
-
-        apiStore = new APIStoreRestClient(storeURLHttp);
-        apiPublisher = new APIPublisherRestClient(publisherURLHttp);
-        apiPublisher.login(user.getUserName(), user.getPassword());
-
-        //adding new role and user
-        userManagementClient1 = new UserManagementClient(keyManagerContext.getContextUrls().getBackEndUrl(),
-                                                         createSession(keyManagerContext));
-        userManagementClient1.addRole(role, null, permissions);
-        userManagementClient1.addUser(allowedUser, String.valueOf(allowedUserPass), new String[] {role }, null);
-        tagListUrl = new URL(getStoreURLHttp() + "store-old/site/blocks/tag/tag-cloud/ajax/list.jag");
+        //Login to API Publisher and Store with CarbonSuper normal user1
+        providerName = publisherContext.getContextTenant().getTenantUser(USER_KEY_USER2).getUserName();
+        User publisherUser1 = publisherContext.getContextTenant().getTenantUser(USER_KEY_USER2);
+        apiPublisherClientCarbonSuperUser1 = new RestAPIPublisherImpl(publisherUser1.getUserNameWithoutDomain(),
+                publisherUser1.getPassword(), publisherUser1.getUserDomain(), publisherURLHttps);
+
+        userManagementClient = new UserManagementClient(keyManagerContext.getContextUrls().getBackEndUrl(),
+                        createSession(keyManagerContext));
+        //add a role for which API tags should be visible
+        userManagementClient.addRole(ROLE, null, PERMISSIONS);
+        // add new user with the above role
+        userManagementClient.addUser(ALLOWED_USER, ALLOWED_USER_PASS, new String[] {ROLE}, null);
+        apiStoreClientAllowedUser = new RestAPIStoreImpl(ALLOWED_USER, ALLOWED_USER_PASS,
+                MultitenantConstants.SUPER_TENANT_DOMAIN_NAME, storeURLHttps);
+        //get a rest api client for anonymous user
+        anonymousRestAPIImpl = super.getRestAPIStoreForAnonymousUser(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
```

Comment:
Do we need to use super here?

Reply:
Fixed in 42613b552de83385cbbf69c553f3333945b20835